### DISCUSSION TITLE
Système d'étoiles à la confirmation

### DISCUSSION THREAD

**Discussion by maximpouliot** [MTESSDev/FRW#112] [Annonces]
À la page de confirmation, nous devons insérer du code qui permet d'afficher un système d'étoile 

![image](https://github.com/MTESSDev/FRW/assets/137327770/476ea575-bfbb-4ac7-9da0-12e6392ca619)

Le code ressemble à ceci : 

```
<a href=https://sondage.retraitequebec.gouv.qc.ca/sondages2/index.php/699177?lang=fr target="_blank" class="rq-retroaction-bloc rq-btn rq-btn-principal avec-icone no-after-ico" rel="noopener noreferrer">
            <span class="d-block d-sm-inline">ÉVALUEZ CE SERVICE</span> 
            <i class="fas fa-star ml-sm-3" aria-hidden="true"></i>
            <i class="fas fa-star" aria-hidden="true"></i>
            <i class="fas fa-star" aria-hidden="true"></i>
            <i class="fas fa-star" aria-hidden="true"></i>
            <i class="fas fa-star" aria-hidden="true"></i> 
        </a>
```

Il faut s'assurer d'avoir le bon css qui est chez nous à Retraite Québec

Comment pourrait-on faire fonctionner ça avec FRW? 

Merci

**Comment by LarcM7** [comment]
Bonjour, nous vous recommandons de vous créer une image en base64 de votre bloc d'étoiles et d'y mettre un lien comme actuellement.

**Reply by jodionrq** [reply]
Bonjour,
Le css fourni qui reproduit l’image avec les étoiles a la particularité de rendre une surbrillance en jaune sur les étoiles lorsqu’on survol (hover) l’image.

Est-ce possible d'intégrer le css dans un code json?
Sinon, est-ce que ça peut être reproduit si on place une image en base 64?

Ce petit css n’est pas l’évaluation en soi, mais un lien vers la page d’évaluation :
[Évaluez nos services – Dépôt direct fait au Canada (gouv.qc.ca)](https://sondage.retraitequebec.gouv.qc.ca/sondages2/index.php/699177?lang=fr%20target=%22_blank%22%20class=%22rq-retroaction-bloc%20rq-btn%20rq-btn-principal%20avec-icone%20no-after-ico%22%20rel=%22noopener%20noreferrer%22)

C'est un exemple que d'une page et le message peut varier selon le service qui a été utilisé par le client, et la langue de préférence (français, anglais).


Si vous avez besoin d'information supplémentaire, vous pouvez me rejoindre aussi par Teams : jonathan.dion@retraitequebec.gouv.qc.ca

**Comment by j0ber** [comment]
Bonjour,
Il n'y a aucun intérêt à rendre une surbrillance en jaune sur les étoiles au survol sachant que ça ne change rien.
Pour moi c'est PIRE, car l'utilisateur va avoir l'impression que s'il clique 2 étoiles, un vote va être enregistré alors que ce n'est pas le cas.
C'est bien ça?

Cela étant dit, il n'est pas du tout dans les plans de permettre l'ajout de css  par les systèmes.

Dans votre cas un simple lien "Répondre à un sondage" (ou autre texte) répondrait déjà au besoin. L'affichage fake de vote n'ajoute rien à part être plus jolie, et comme mentionné précédemment, c'est trompeur.

**Reply by jodionrq** [reply]
Bonjour, merci pour la réponse!
Si on veut intégrer ce lien vers le sondage dans tous nos formulaires, est-ce possible de programmer ce lien dans le fichier Default et de placer une variable ou quelque chose qui renvoie vers ce lien? De cette façon, s'il y a une mise à jour, on aurait seulement un endroit à modifier. Et si c'est possible, avez-vous un exemple?

**Comment by LarcM7** [comment]
Bonjour,

Oui vous pouvez personnaliser le texte affiché sur la page de confirmation de transmission dans votre fichier default et il s'appliquera à tous vos formulaires.

Les pages suivantes du P700 sont pertinentes, et contiennent des exemples : 
- https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=personnalisationBase
- https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=pageConfirmation

**Reply by jodionrq** [reply]
Merci pour le réponse!
Donc, si je place mon truc d'évaluation dans la confirmationTransmission de mon fichier default, et que je place aussi du texte supplémentaire dans la config confirmationTransmission d'un formulaire, tout devrait s'afficher?

**Reply by LarcM7** [reply]
Bonjour,
Non. 

C'est expliqué dans le P700 dans la page "[Concepts de base](https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=personnalisationBase)". 😊

> **Hiérarchie dans la personnalisation**
> Lorsqu'une même clé de configuration est définie dans plusieurs configurations, le niveau le plus précis a toujours préséance sur les autres.

J'espère que ça répond à votre question, n'hésitez pas s'il y a quoi que ce soit.

**Reply by jodionrq** [reply]
Bonjour, merci pour la précision, c'est bien ce que je pensais.
Donc, si le texte de confirmation de transmission change d'un formulaire à l'autre, il faut intégré le truc d'évaluation dans la config confirmationTransmission de chaque formulaire.
Si nous plaçons une image en base 64, ça fait un gros bloc de code et il faudrait le mettre à jour partout s'il y a une modification.
Est-ce qu'il y a un moyen d'aller chercher un lien vers une image source placé sur un serveur? Idéalement ce lien ne devrait pas changé lorsqu'il y a une mise à jour bien sûr.
Avez-vous des cas où un élément doit être présent dans plusieurs formulaires, qui ne se trouve pas dans le "default" mais qui risque d'être mis à jour partout, et si oui, comment le gérez-vous?

**Reply by LarcM7** [reply]
Bonjour,

> Donc, si le texte de confirmation de transmission change d'un formulaire à l'autre, il faut intégré le truc d'évaluation dans la config confirmationTransmission de chaque formulaire.

Dans l'état actuel, oui. 

Est-ce que ça répondrait à votre besoin s'il était possible de personnaliser 2 lignes dans confirmationTransmission au lieu d'une seule? De façon indépendante par niveau. Par exemple : 

Personnalisation au niveau système (texteSupplementaire1) :
```
config:  
  confirmationTransmission:
    texteSupplementaire1:
      fr:  <votre système d'étoile> 
      en: 
```
Personnalisation au niveau formulaire (texteSupplementaire2) :
```
config:  
  confirmationTransmission:
    texteSupplementaire2:
      fr:  <un texte personnalisé en fonction de ce formulaire en particulier> 
      en: 
```
Si oui, vous pouvez faire une demande pour cette fonctionnalité via les issues.

> Si nous plaçons une image en base 64, ça fait un gros bloc de code et il faudrait le mettre à jour partout s'il y a une modification. 
> Est-ce qu'il y a un moyen d'aller chercher un lien vers une image source placé sur un serveur? Idéalement ce lien ne devrait pas changé lorsqu'il y a une mise à jour bien sûr.

Je pense que la proposition précédente répondrait à ce besoin.

> Avez-vous des cas où un élément doit être présent dans plusieurs formulaires, qui ne se trouve pas dans le "default" mais qui risque d'être mis à jour partout, et si oui, comment le gérez-vous?

Nous avions déjà reçu une demande similaire. Je viens de l'ouvrir ici dans GitHub https://github.com/MTESSDev/FRW/issues/127. Est-ce que celle-ci répondrait à votre besoin? Vous pouvez commenter.

**Reply by jodionrq** [reply]
Bonjour, merci pour la réponse et pour vos suggestions!
Je vais regarder ça avec mon équipe et je vous reviens.

En attendant, si on prend la première proposition, est-ce que le texteSupplémentaire1 se placerait toujours avant le texteSupplémentaire2 ? Si mon truc d'évaluation serait au niveau système, est-ce qu'il peut apparaitre après le texte personnalisé au niveau formulaire?

**Reply by LarcM7** [reply]
Oui le texte 1 serait avant le 2.
Oui, vous n'auriez qu'à mettre votre truc dans le texte 2 au niveau système, et personnaliser le texte 1 au niveau du formulaire!
